src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -1,26 +1,33 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-using NuGet.Services.Metadata.Catalog.Persistence;
+
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Helpers;
+using NuGet.Services.Metadata.Catalog.Persistence;
 using VDS.RDF;
 
 namespace NuGet.Services.Metadata.Catalog.Registration
 {
     public class RegistrationCollector : SortingGraphCollector
     {
-        StorageFactory _storageFactory;
+        private readonly StorageFactory _semVer1StorageFactory;
+        private readonly StorageFactory _semVer2StorageFactory;
 
-        public RegistrationCollector(Uri index, StorageFactory storageFactory, Func<HttpMessageHandler> handlerFunc = null)
+        public RegistrationCollector(
+            Uri index,
+            StorageFactory semVer1StorageFactory,
+            StorageFactory semVer2StorageFactory,
+            Func<HttpMessageHandler> handlerFunc = null)
             : base(index, new Uri[] { Schema.DataTypes.PackageDetails, Schema.DataTypes.PackageDelete }, handlerFunc)
         {
-            _storageFactory = storageFactory;
+            _semVer1StorageFactory = semVer1StorageFactory;
```

Comment:
Legacy seems right. However I think our the definition in NuGet world is always going to be blurred. For example `1.0.0.1-alpha.1` is considered "SemVer 2.0.0" by NuGet.